### PR TITLE
Fix gas checkpoint resuming

### PR DESCRIPTION
**Pull request by alexandonian**
Despite best efforts to fix checkpoint resuming, we've identified an additional bug that occurs when resuming a run with `gradient_accumulation_steps` set to a value above one. In particular, a training run that required continual restarts due to job time limits began producing the following loss trajectory:

gas_bug

Clearly, the run is not resuming in the exact same state as it left off, and small scale experiments isolated the issue the gas > 1 setting. The loss curves suggests that the model was revisiting data it had just previously encountered before the checkpoint. Indeed, upon resuming, the dataloader start iteration was not being fast forwarded in a manner that took into account the gradient accumulation steps, and this PR addresses this issue. This update appears to have restored the step-for-step perfect checkpoint resuming as seen below:

gas_fix

### CODE REVIEW

**Review by kingoflolz — DISMISSED** [review]
Looks good, could you remove the whitespace/formatting changes though?

**Comment by sdtblck** [comment]
Also we need to

- Make the same changes for validation dataloader resuming
- Make sure non pipe parallel training is actually respecting g.a.s (Currently i don't think it is). Would be good to make all these changes in one PR.

**Comment by sdtblck** [comment]
Ok, I made the changes above. Think it should be good to merge now.

**Review by sdtblck — APPROVED** [review]
